frontends/web/src/components/devices/bitbox02/bitbox02.tsx

```diff
@@ -349,19 +355,13 @@ class BitBox02 extends Component<Props, State> {
         });
     }
 
-    private setDisclaimerRef = (element: HTMLElement) => {
-        this.disclaimerForm = element;
-    }
-
-    private handleDisclaimerCheck = () => {
-        const checkboxes = this.disclaimerForm.querySelectorAll('input');
-        let result = true;
-        for (const checkbox of checkboxes) {
-            if (!checkbox.checked) {
-                result = false;
-            }
-        }
-        this.setState({ readDisclaimers: result });
+    private handleDisclaimerCheck = (event: InputEvent) => {
+        const target = event.target as HTMLInputElement;
+        const name = target.id as 'argement1' | 'argement2' | 'argement3' | 'argement4' | 'argement5';
+        this.setState(prevState => ({
+            ...prevState,
```

Comment:
i will never understand what this does and what a simple `this.setState({ [name]: target.checked })` would do wrong here. Can you enlighten me? :D

Reply:
The normal reason why you would do that is if you want to change the state based on the **current state**, i.e. toggle something on/off, this would be important if you click faster than react updating the state. In that case we can pass an async function instead of an object. 

see 
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

```js
// bad
this.setState({
  counter: this.state.counter + this.props.increment,
});

// good
this.setState((state, props) => ({
  counter: state.counter + props.increment
}));
```

but in this particular case, it is because TypeScript didn't like it... I didn't really deeply investigate deeply why 🙈

Reply:
same issue https://stackoverflow.com/questions/42090191/picks-k-type-with-dynamic-computed-keys

Reply:
It seems Typescript doesn't like ES6 style dynamic property keys, but doesn't complain about oldschool

```js
const obj = {};
obj[target.id] = target.checked;
this.setState(obj);
```